### PR TITLE
Add metro support for equinixmetal

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
Adding support for equinixmetal metro field in machine controller. This is needed to enable machine controller equinixmetal cloud provider to pick any server available in a whole metro(group of facilities) to avoid availability problems when deploying multiple replica. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
Equinixmetal metro support in machine controller 
```

### CODE REVIEW

File: pkg/cloudprovider/provider/equinixmetal/types/types.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type RawConfig struct {
 	ProjectID    providerconfigtypes.ConfigVarString   `json:"projectID,omitempty"`
 	BillingCycle providerconfigtypes.ConfigVarString   `json:"billingCycle"`
 	InstanceType providerconfigtypes.ConfigVarString   `json:"instanceType"`
+	Metro        providerconfigtypes.ConfigVarString   `json:"metro"`
```
Comment (ahmedwaleedmalik) [inline pkg/cloudprovider/provider/equinixmetal/types/types.go:29]: Shouldn't this be an optional field? Basically, both Facilities and metro should be made optional and then we can check in `Validate` that at least one of them is specified. WDYT?

Comment (moadqassem) [reply pkg/cloudprovider/provider/equinixmetal/types/types.go:29]: I do this already https://github.com/kubermatic/machine-controller/pull/1339/files#diff-86dce1d2d7e8fb9dfddab19175ce5b62bb771587033f0bb6a3b83961b9318e75R195

Comment (moadqassem) [reply pkg/cloudprovider/provider/equinixmetal/types/types.go:29]: Oh actually that's true, will adjust that now

**Pull request merged**